### PR TITLE
feat: v2 customer content-metadata endpoint

### PR DESCRIPTION
**Pull request by iloveagent57**
## Testing
You need a restricted run setup first, see for instructions: https://github.com/openedx/enterprise-catalog/pull/985

### Content Metadata Retrieve
```
GET http://localhost:18160/api/v2/enterprise-customer/378d5bf0-f67d-4bf7-8b2a-cbbc53d0f772/content-metadata/course-v1%3AedX%2BDemoX%2Brestricted_run/
```
response:
```
{
    "key": "course-v1:edX+DemoX+restricted_run",
    "uuid": "4772f672-583d-401a-a948-34b1a66e9cde",
    "title": "edX Demonstration Course",
    "external_key": null,
    "fixed_price_usd": null,
    "image": null,
    "short_description": "",
    "marketing_url": "course/test-comrsch-ajedrgwvydvk-course-v1orgsuytibiasoigfake?utm_medium=enterprise&utm_source=pied-piper",
    "seats": [
        {
            "type": "verified",
            "price": "73.10",
            "currency": "AED",
            "upgrade_deadline": "2028-10-21T20:27:24Z",
            "upgrade_deadline_override": "2028-10-21T20:27:24Z",
            "credit_provider": null,
            "credit_hours": null,
            "sku": "RxucNNGN",
            "bulk_sku": "DgJCzVNo"
        }
    ],
    "start": "2015-02-02T02:46:43Z",
    "end": "2027-07-09T17:13:41Z",
    "go_live_date": null,
    "enrollment_start": "2016-07-12T06:48:26Z",
    "enrollment_end": "2027-09-10T06:33:49Z",
    "weeks_to_complete": 1,
    "pacing_type": "instructor_paced",
    "type": "verified",
    "restriction_type": "custom-b2b-enterprise",
    "run_type": "f1340c0d-67af-40c1-9179-f209f5bf9515",
    "status": "unpublished",
    "is_enrollable": true,
    "is_marketable": false,
    "availability": "Current",
    "variant_id": null,
    "course": "edX+DemoX"
    # there's more, truncated for brevity
}
```

### Contains Content Items
```
GET http://localhost:18160/api/v2/enterprise-customer/378d5bf0-f67d-4bf7-8b2a-cbbc53d0f772/contains_content_items/?course_run_ids=course-v1%3AedX%2BDemoX%2Brestricted_run&get_catalog_list=true
```
response:
```
{
    "contains_content_items": true,
    "catalog_list": [
        "7467c9d2-433c-4f7e-ba2e-c5c7798527b2"
    ]
}
```

### Filter content items
```
POST http://localhost:18160/api/v2/enterprise-customer/378d5bf0-f67d-4bf7-8b2a-cbbc53d0f772/filter_content_items/

{
    "content_keys": ["course-v1:edX+DemoX+restricted_run", "course-v1:some+other+run"]    
}
```
response
```
{
    "filtered_content_keys": [
        "course-v1:edX+DemoX+restricted_run"
    ]
}
```
## Post-review

* [ ] Squash commits into discrete sets of changes
* [ ] Ensure that once the changes have been [deployed to stage](https://gocd.tools.edx.org/go/pipeline/activity/stage-enterprise_catalog), prod is manually deployed

### CODE REVIEW

File: enterprise_catalog/apps/api/v1/tests/test_enterprise_customer_views.py [hunk 1]
```diff
@@ -357,152 +302,6 @@ def test_contains_catalog_list_with_catalog_list_param(self):
         catalog_list = response.json()['catalog_list']
         assert set(catalog_list) == {str(second_catalog.uuid)}
 
-    @pytest.mark.skip(reason="We need a version of this test for the v2 API.")
```
Comment (iloveagent57) [inline enterprise_catalog/apps/api/v1/tests/test_enterprise_customer_views.py:360]: These tests were all written with our now-abandoned restricted run architecture in mind.

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py [hunk 2]
```diff
@@ -164,17 +175,15 @@ def get_metadata_item_serializer(self):
             # identifier is a valid UUID.
             content_uuid = uuid.UUID(content_identifier)
             for catalog in enterprise_catalogs:
-                content_with_uuid = catalog.content_metadata.filter(content_uuid=content_uuid)
-                if content_with_uuid:
+                if content_with_uuid := self.get_metadata_by_uuid(catalog, content_uuid):
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/api/v1/views/enterprise_customer.py:178]: nit/curious: was a bit thrown off by the naming of this method `self.get_metadata_by_uuid`, expecting it to return a single instance vs a list given the `get` prefix (similar for `self.get_metadata_by_content_key`, too). A couple thoughts:

* would it potentially make sense to rename these helper methods to `list_metadata_by_uuid` and `list_metadata_by_content_key`?
* alternatively, is there an opportunity to call the `.first()` within the helper methods themselves so they do return a single instance?

File: enterprise_catalog/apps/api/v2/tests/test_enterprise_customer_views.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        catalog = EnterpriseCatalogFactory(enterprise_uuid=self.enterprise_uuid)
+        catalog_b = EnterpriseCatalogFactory(enterprise_uuid=self.enterprise_uuid)
+
+        content_one, _, restricted_run = self._create_restricted_course_and_run(catalog)
+
+        self.add_metadata_to_catalog(catalog, [content_one, restricted_run])
+        self.add_metadata_to_catalog(catalog_b, [content_one])
+
+        url = self._get_contains_content_base_url() + \
+            f'?course_run_ids={restricted_run.content_key}&get_catalog_list=true'
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/api/v2/tests/test_enterprise_customer_views.py]: [nit/curious] Do we want to continue using the query parameter `get_catalog_list` vs. `get_catalogs_containing_specified_content_ids`? In the enterprise-catalog docstring around these params, it denotes `get_catalog_list` as being "old" (legacy/deprecated?).

https://github.com/openedx/enterprise-catalog/blob/4722f069ec7916be23ea3a4da85e049c83d3c5c0/enterprise_catalog/apps/api/v1/views/enterprise_customer.py#L80-L85

File: enterprise_catalog/apps/api/v2/views/enterprise_customer.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        return catalog.content_metadata_with_restricted.filter(content_uuid=content_uuid)
+
+    def get_metadata_by_content_key(self, catalog, content_key):
+        return catalog.get_matching_content(content_keys=[content_key], include_restricted=True)
+
+    def filter_content_keys(self, catalog, content_keys):
+        return catalog.filter_content_keys(content_keys, include_restricted=True)
+
+    def contains_content_keys(self, catalog, content_keys):
+        return catalog.contains_content_keys(content_keys, include_restricted=True)
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/api/v2/views/enterprise_customer.py]: Nice job with the inheritance; just having to override/extend these methods in the V2 viewset is pretty clean :)

**Review by adamstankiewicz — APPROVED** [review]
Overall, looks great! I just had a couple nit comments to clarify/consider.

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py (hunk 2 above)
Comment (iloveagent57) [reply enterprise_catalog/apps/api/v1/views/enterprise_customer.py:178]: ah, good call, I moved the `.first()` call into the helper methods, much cleaner that way.

**Pull request merged**